.github/actions/cache-evm-artifacts/action.yml

```diff
@@ -11,14 +11,19 @@ outputs:
 runs:
   using: "composite"
   steps:
+    - name: Resolve Node version
+      id: resolved-node
+      uses: ./.github/actions/setup-node-if-needed
+      with:
+        node_version: ${{ inputs.node_version }}
     - name: Restore cached EVM artifacts
       id: evm-artifacts-cache
       uses: actions/cache@v4
       with:
         # The job that generates the artifacts is responsible for archiving them to the cache tarball. This avoids any
         # conflicts with other caching actions that might have cleaned some of cached contents.
         path: evm-artifacts.tar
-        key: evm-artifacts-${{ runner.os }}-node-${{ inputs.node_version }}-${{ hashFiles('yarn.lock', 'hardhat.config.ts', 'contracts/**/*.sol') }}
+        key: evm-artifacts-${{ runner.os }}-node-${{ steps.resolved-node.outputs.version }}-${{ hashFiles('yarn.lock', 'hardhat.config.ts', 'contracts/**/*.sol') }}
```

Comment:
This makes it possible to reuse cache on release when it was generated during the latest push to master